angularSTEP/src/app/item-dialog/item-dialog.component.ts

```diff
@@ -0,0 +1,42 @@
+import {Component, Inject} from '@angular/core';
+import {FormBuilder, FormControl, FormGroup, Validators} from '@angular/forms';
+import {MAT_DIALOG_DATA, MatDialogRef} from '@angular/material/dialog';
+
+@Component({
+  selector: 'app-item-dialog',
+  templateUrl: './item-dialog.component.html',
+  styleUrls: ['./item-dialog.component.scss']
+})
+export class ItemDialogComponent {
+  form: FormGroup = new FormGroup({
+    quantity: new FormControl(1, Validators.required),
+  })
+  selectedItem: string;
+  add: boolean;
+  remove: boolean;
+  max: number;
+
+  constructor(
+    private fb: FormBuilder,
+    private dialogRef: MatDialogRef<ItemDialogComponent>,
```

Comment:
I guess this somehow wires up the `this.dialogRef` magic? That's my running theory anyhow